### ISSUE TITLE
github building with out-of-date Visual Studio

### ISSUE THREAD

**Issue by wsu-cb** [CyberBoardPBEM/cbwindows#117]
Dale,
In working on #116, I have discovered that the Visual Studio that we are being given by github has an out-of-date STL.  I have added a commit to work around the problem, but it would be nice if we could use an up-to-date Visual Studio.  Do you know how to get the current MSVS from github?

**Comment by DLLarson** [comment]
> Do you know how to get the current MSVS from github?

I'll check on this later today.
-Dale

**Comment by DLLarson** [comment]
It looks like the notes for the updated VS 2022 software (17.6) was just posted today:

https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes#17.6.0

I just updated my own VS2022 to 17.6 today. I had to tell VS to check for updates as it didn't pop up a message when I loaded the project like it usually does when I get an update notice.

The program data for the workflow stuff that we're running as still lists the previous release 17.5.somthing:

https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md

Although they may just not update the notes right away.

-Dale

**Comment by wsu-cb** [comment]
The problem is fixed by https://github.com/microsoft/STL/commit/67a96a910fb53aa4bc12801fb9683d30b2647c16 .  The strange thing is that my MSVS says it's 17.5.1, but it has this fixed already.  I don't understand why github and I have different STL versions for the same MSVS version, nor do I see a way to identify exactly what STL version a particular system has.

**Comment by DLLarson** [comment]
>The strange thing is that my MSVS says it's 17.5.1, but it has this fixed already. I don't understand why github and I have different STL versions for the same MSVS version, nor do I see a way to identify exactly what STL version a particular system has.

Very weird. I'm rebuilding all targets (including the 'U' variants) on my dev machine and everything builds fine.

If I run the VS command line environment I see this for the version:

![image](https://github.com/CyberBoardPBEM/cbwindows/assets/175455/da2a106b-2ed5-4f6f-bd23-d6244408fb43)

Can you check what you see?

I also [found this](https://en.cppreference.com/w/cpp/utility/format). It makes the situation kinda muddy:

![image](https://github.com/CyberBoardPBEM/cbwindows/assets/175455/a6b318c0-62f2-4ca8-9376-bb47d5ea8a90)

-Dale

**Comment by wsu-cb** [comment]
g:\cyberboard\github\cbwindows>msbuild -ver
MSBuild version 17.5.0+6f08c67f3 for .NET Framework
17.5.0.10706

**Comment by DLLarson** [comment]
I'm out of ideas. 

This would mean your STL source has `__cpp_lib_format` defined as `202207L`?

Unfortunately my system has been upgraded and I don't think downgrades are possible.

-Dale

**Comment by wsu-cb** [comment]
> I'm out of ideas.

I don't understand this discrepancy either, but since I have found a workaround, I don't think it's a serious concern.  On the other hand, this is definitely something that makes one wonder what is really going on.

> This would mean your STL source has `__cpp_lib_format` defined as `202207L`?

Yes

**Comment by DLLarson** [comment]
Well let's just ride it out for a bit to see if the GitHub build environment gets updated.

I take it for the moment I should wait to do the code review thingy on your latest UNICODE commit?

-Dale

**Comment by wsu-cb** [comment]
> I take it for the moment I should wait to do the code review thingy on your latest UNICODE commit?

Yes and no.  I pushed it to preserve the state because I think it has reached a valuable milestone, but on other hand, the implementation is using what I expect to be a temporary method of converting between utf8 and wchar_t.  The real intent is to add wxWidgets, and use it to deal with those conversions.  Unfortunately, I am running into trouble with the combination of wxWidgets, MFC, and static CRT.  I have posted a question to the wxWidgets discussion list about that.
